src/java/com/zimbra/oauth/handlers/impl/FacebookOAuth2Handler.java

```diff
@@ -130,27 +132,31 @@
         /**
          * The authorize uri template for Facebook.
          */
-        protected static final String AUTHORIZE_URI_TEMPLATE = "https://www.facebook.com/v3.0/dialog/oauth?client_id=%s&redirect_uri=%s&response_type=%s&scope=%s";
+        protected static final String AUTHORIZE_URI_TEMPLATE =
+            "https://www.facebook.com/v3.0/dialog/oauth"
+            + "?client_id=%s&redirect_uri=%s&response_type=%s&scope=%s";
 
         /**
-         * The debug token uri, used to fetch the user_id from Facebook.
+         * The user details uri, used to fetch the authorized OAuth user details from Facebook.
          */
-        public static final String DEBUG_TOKEN_URI = "https://graph.facebook.com/debug_token";
+        public static final String USER_DETAILS_URI_TEMPLATE = "https://graph.facebook.com/me"
+            + "?access_token=%s&fields=first_name,middle_name,last_name,email";
 
         /**
          * The uri used to make backend call to fetch an access token.
          */
-        public static final String AUTHENTICATE_URI = "https://graph.facebook.com/v3.0/oauth/access_token";
+        public static final String AUTHENTICATE_URI =
```

Comment:
Make all constants `protected`.

The package classes that need it have access.